docs/specification.md

```diff
@@ -311,89 +432,153 @@ available state entry is `D`:
+Note that as invalid RSL entries are only created by buggy or malicious gittuf
+clients, these entries cannot be detected prior to them being pushed to the main
+repository.
+
+As correctly implemented gittuf clients verify the validity of RSL entries when
+they pull from the main repository, the user is warned if invalid entries are
+encountered. Then, the user can then use M1 to invalidate the incorrect entry.
+Other clients with the invalid entry only need to fetch the latest RSL entries
+to recover. The client that originally created the invalid entries must first
+switch to a correct implementation of gittuf and then sync the local
+repository's state with the main repository.
+
+If the main repository is also gittuf enabled, such incidents can be caught
+before other users receive the incorrect RSL entries. The repository, though,
+must not behave like a typical gittuf client. Instead, gittuf's repository
+behavior is slightly different as RSL entries are submitted before the changes
+they represent. The repository must wait to receive the full changes rather than
+immediately rejecting the RSL entry. TODO: the repository-specific behavior
+needs further discussion.
+
+Consider this example as a representative of this scenario. Bob has a buggy
+gittuf client and pushes an invalid entry to the main repository. Alice pulls
+and receives a warning from her gittuf client. Alice reverses Bob's changes,
+creating an RSL entry for the affected branch if needed, and includes an RSL
+annotation skipping Bob's RSL entry.
+
+#### A gittuf access control policy is violated
+
+Bob creates an RSL entry for a branch he's not authorized for by gittuf policy.
+He pushes a change to that branch. Alice notices this (TODO: decide if alice
+needs to be authorized). Alice reverses Bob's change, creating a new RSL entry
+reflecting that. Alice also creates an RSL annotation marking Bob's entry as
+one to be skipped. Alice, therefore, uses M1.
+
+#### Attacker modifies or deletes historical RSL entry
+
+Overwriting or deleting an historical RSL entry is a complicated proposition.
+Git's content addressable properties mean that a SHA-1 collision is necessary to
+overwrite an existing RSL entry in the Git object store. Further, the attacker
+also needs more than push access to the repository as Git will not accept an
+object it already has in its store (TODO: double check). Similarly, deleting an
+entry from the object store preserves the RSL structure cosmetically but
+verification workflows that require the entry will fail.
+
+Also note that while Git uses SHA-1 for its object store, cryptographic 
+signatures are generated and verified using stronger hash algorithms. Therefore,
+a successful SHA-1 collision for an RSL entry will not go undetected as all
+entries are signed.
+
+#### Dealing with fork* attacks
+
+An attacker may attempt a forking attack where different developers receive
+different RSL states. This will not succeed, however, against any developers
```

Comment:
What does it mean to succeed in this context?

Reply:
Updated.